src/Sentry/GlobalSessionManager.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Sentry.Extensibility;
+using Sentry.Internal;
+
+namespace Sentry
+{
+    // AKA client mode
+    internal class GlobalSessionManager : ISessionManager
+    {
+        private readonly object _lock = new();
+        private readonly SentryOptions _options;
+
+        private string? _cachedInstallationId;
+        private Session? _currentSession;
+
+        public GlobalSessionManager(SentryOptions options)
+        {
+            _options = options;
+        }
+
+        private string GetInstallationId()
+        {
+            // Prevent race conditions that could cause us to generate multiple IDs.
+            // Note: this has to be synchronized across multiple processes too.
+            lock (_lock)
+            {
+                // Avoid IO if possible
+                if (!string.IsNullOrWhiteSpace(_cachedInstallationId))
+                {
+                    return _cachedInstallationId;
+                }
```

Comment:
avoid taking the lock too: double check locking

```suggestion
            if (!string.IsNullOrWhiteSpace(_cachedInstallationId))
            {
                return _cachedInstallationId;
            }

            // Prevent race conditions that could cause us to generate multiple IDs.
            // Note: this has to be synchronized across multiple processes too.
            lock (_lock)
            {
                // Avoid IO if possible
                if (!string.IsNullOrWhiteSpace(_cachedInstallationId))
                {
                    return _cachedInstallationId;
                }
```